pkg/k8s/apis/cilium.io/v2/types.go

```diff
@@ -458,3 +459,43 @@ func (n *CiliumNode) InstanceID() (instanceID string) {
 	}
 	return
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:singular="ciliumegressgatewaypolicy",path="ciliumegressgatewaypolicies",scope="Cluster"
+// +kubebuilder:printcolumn:JSONPath=".metadata.labels.io\\.kubernetes\\.pod\\.namespace",description="The namespace of the entity",name="Namespace",type=string
+// +kubebuilder:printcolumn:JSONPath=".metadata.creationTimestamp",description="The age of the identity",name="Age",type=date
+// +kubebuilder:subresource:status
+// +kubebuilder:storageversion
+
+type CiliumEgressGatewayPolicy struct {
+	// +k8s:openapi-gen=false
+	// +deepequal-gen=false
+	metav1.TypeMeta `json:",inline"`
+	// +k8s:openapi-gen=false
+	// +deepequal-gen=false
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec CiliumEgressGatewayPolicySpec `json:"spec,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=false
+// +deepequal-gen=false
+
+// CiliumEgressGatewayPolicyList is a list of EgressGatewayPolicy objects.
+type CiliumEgressGatewayPolicyList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	// Items is a list of CiliumEgressGatewayPolicy
+	Items []CiliumEgressGatewayPolicy `json:"items"`
+}
+
+type CiliumEgressGatewayPolicySpec struct {
+	PodSelector       *slimv1.LabelSelector `json:"podSelector,omitempty"`
+	NamespaceSelector *slimv1.LabelSelector `json:"namespaceSelector,omitempty"`
+	DestinationCIDRs  []string              `json:"destinationCidrs,omitempty"`
+	EgressSourceIp    string                `json:"egressSourceIp,omitempty"`
```

Comment:
Another suggestion for `EgressSourceIp` would be to copy-paste the validation kubebuilder regex for IPs  from another CRD which has an IP field.

Reply:
Fixed. Thanks!